### ISSUE TITLE
Provide generic template value for registerConstructor function

### ISSUE THREAD

**Issue by eigan** [CuyZ/Valinor#149]
_Sorry for all the issues, I really love the library!_ 😅 

Might be related to #8, though that was when `@template` was used at root.

```php
<?php

require "vendor/autoload.php";

class FooBar {
    public function __construct(public int $id) {}
}

/**
 * @template T
 */
class Bar
{
    /**
     * @var T
     */
    public $obj;

    /**
     * @param T $obj
     */
    public function __construct($obj) {
        $this->obj = $obj;
    }
}

class Foo
{
    /**
     * @var Bar<FooBar>
     */
    public Bar $bar;
}

$mapper = (new \CuyZ\Valinor\MapperBuilder())
    ->registerConstructor(

        /**
         * @template T
         * @param class-string<T> $className
         * 
         * @return Bar<T>
         */
        #[\CuyZ\Valinor\Mapper\Object\DynamicConstructor]
        function ($className, $id): Bar {
            $obj = new $className($id);
            return new Bar($obj);
        }
    )
    ->mapper();

$foo = $mapper->map(Foo::class, [
    'bar' => 2
]);

print $foo->bar->obj->id;

```

- This is not a real-life example

**Comment by romm** [comment]
Hi @eigan, I might have not understood well your issue, but the following works in Valinor:

```php
class FooBar
{
    public function __construct(public int $id) {}
}

/**
 * @template T
 */
class Bar
{
    /**
     * @var T
     */
    public $obj;

    /**
     * @param T $obj
     */
    public function __construct($obj)
    {
        $this->obj = $obj;
    }
}

class Foo
{
    /**
     * @var Bar<FooBar>
     */
    public Bar $bar;
}

$mapper = (new \CuyZ\Valinor\MapperBuilder())->mapper();

$foo = $mapper->map(Foo::class, [
    'bar' => 2
]);
```

You don't have a register a "generic" constructor in this case, the library will parse and analyse `@template` annotations just right.

**Comment by romm** [comment]
Hi @eigan, as you closed #354 I'll now close this issue. Please feel free to reopen or post a new issue if your issue persists.

Have a good day!
